EIPS/eip-7779.md

```diff
@@ -0,0 +1,106 @@
+  
+- **Automatic Deactivation:**
+  - A safeguard deactivation block will halt the increase after two years, ensuring that the gas limit does not grow indefinitely without further consensus from the community.
+
+## **Specification**
+
+### **Incremental Gas Limit Increase**
+
+1. **Initial Block Gas Limit:**
+   - At the start of the proposal, the block gas limit is set to 30 million gas.
+
+2. **Increase Rate:**
+   - The block gas limit will increase by 0.00475% per block (approximately 1.5% per day), such that after approximately 730 days (two years), the gas limit reaches 60 million.
+
+3. **Deactivation Block:**
+   - The gas limit increase will automatically halt at a specific block height, calculated as the block occurring roughly two years after the proposal's implementation. This "deactivation block" ensures no further increases occur after this period.
+
+4. **Formula for Gas Limit Increase:**
+   - The gas limit `GL_t` at block `t` is calculated as:
+   ```python
+   GL_t = GL_0 * (1 + r)^t
+   ```
+   Where:
+   - `GL_0` is the initial gas limit (30 million gas).
+   - `r` is the per-block increase rate (0.00475% or 0.0000475).
+   - `t` is the number of blocks since the implementation of the EIP.
+   - The increase halts once `t` reaches the block corresponding to two years (approximately 5,184,000 blocks, assuming 12-second block times).
+
+5. **Post-Deactivation Block Limit:**
+   - After the deactivation block, the gas limit remains fixed at 60 million gas unless further action is taken by the community.
+
+### **Implementation**
+
+- **Client Updates:**
+  - Ethereum clients must implement the automatic gas limit increase according to the specified rate and formula, with enforcement of the deactivation block.
+  
+- **Gas Limit Management:**
+  - Miners and validators are not required to adjust the gas limit manually during this period, as the increase will be automated at the protocol level.
+
+## **Rationale**
+
+### **Predictable Growth**
+
+- **Systematic Adjustment:**
+  - The gradual increase avoids sudden surges in gas limit that could destabilize the network. Instead, it provides a smooth transition, giving the ecosystem time to adapt to larger block sizes.
+
+### **Controlled Limit with Deactivation Block**
+
+- **Automatic Safeguard:**
+  - The inclusion of a deactivation block ensures that the gas limit does not increase indefinitely, preventing potential negative impacts on network performance beyond the planned growth.
+
+- **Post-Deactivation Flexibility:**
+  - After the two-year period, the community can reassess the network's needs and decide on further gas limit adjustments if required.
+
+- **Community Consensus:**
+  - The deactivation block serves only as a checkpoint for the community to evaluate the impact of the gas limit increase, however, in the two-year period, the community can decide to halt the increase at any time as part of any hard fork during the two-year period.
```

Comment:
This halting process will take a very long time. If it is decided that this EIP should be paused, in order to get all clients on board for the fork and the testing, even if the fork is only this EIP it will likely take weeks. (Which is problematic if we absolutely need to pause this mechanism)

Reply:
Fixed your concerns in the updated version of the EIP